File: Bai_112/main.py
k = int(input())
intmax = 1000000
arr = [1] * intmax
curr = 2
res = 0

while curr < intmax:
    if arr[curr] == 1:
        temp = curr
        while temp < intmax:
            arr[temp] = 0
            temp += curr
    curr += 1

arr1 = list(map(int, input().split()))
for i in arr1:
    res += 1 if arr[i] == 1 else 0

print(res)

File: Bai_112/main.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int k; cin >> k;
    const int intmax = 1000000;
    vector<int> arr(intmax, 1);
    int curr = 2, res = 0;

    // Sàng Eratosthenes
    while(curr < intmax) {
        if(arr[curr] == 1){
            for(int temp = curr * 2; temp < intmax; temp += curr)
                arr[temp] = 0;
        }
        curr++;
    }

    vector<int> arr1(k);
    for(int i = 0; i < k; ++i) cin >> arr1[i];
    for (int i = 0; i < k; ++i)
        if (arr[arr1[i]] == 1) 
            res++;
    cout << res << endl;
    return 0;
}
